src/components/steps.tsx

```diff
@@ -0,0 +1,91 @@
+import { CheckCircleIcon } from '@chakra-ui/icons';
+import { GridProps, Grid, Icon, Text, Box, IconProps } from '@chakra-ui/react';
+import { ReactNode, SVGProps } from 'react';
+
+interface Item {
+  label: ReactNode;
+  value: string | number;
+}
+
+interface Props extends GridProps {
+  itemList: Item[];
+  activeItemValue?: string | number;
+}
+
+function CircleSvg(props: SVGProps<SVGCircleElement>) {
+  return (
+    <circle
+      {...props}
+      cx="16"
+      cy="16"
+      r="15"
+      fill="transparent"
+      stroke="currentColor"
+      strokeWidth={2}
+    />
+  );
+}
+
+function StepIcon(props: IconProps) {
+  return (
+    <Icon {...props} color="secondary.100" viewBox="0 0 32 32">
+      <CircleSvg />
+    </Icon>
+  );
+}
+
+function StepIconActive(props: IconProps) {
+  return (
+    <Icon {...props} color="secondary.100" viewBox="0 0 32 32">
+      <CircleSvg />
+      <circle cx="16" cy="16" r="5" fill="currentColor" />
+    </Icon>
+  );
+}
+
+function StepIconCompleted(props: IconProps) {
+  return <Icon {...props} as={CheckCircleIcon} color="green.100" />;
+}
+
+export default function Steps({ itemList, activeItemValue, ...props }: Props) {
+  const activeItemIndex = itemList.findIndex(
+    (item) => item.value === activeItemValue,
+  );
+
+  const itemListNodes = itemList.flatMap((item, i) => {
+    const isActive = i === activeItemIndex;
+    const isCompleted = i < activeItemIndex;
+
+    let IconComponent = StepIcon;
```

Comment:
59-61 I'd put into function `IconComponent` which takes `isActive` and `isCompleter` as arguments.
and use early return to avoid else-if